### PR TITLE
[BUGFIX] Failures in `pivot_wider`

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- handle deprecation warnings where possible
-  use vectorized option for truncate_datetime
- fix failing tests in pivot_wider

**This PR improves pivot_wider.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.md`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by ericmjl** [comment]
🚀 Deployed on https://deploy-preview-1337--pyjanitor.netlify.app

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 -   [ENH] `change_index_dtype` added. - @samukweku Issue #1314
 -   [ENH] Add `glue` and `axis` parameters to `collapse_levels`. - Issue #211 @samukweku
 -   [ENH] `row_to_names` now supports multiple rows conversion to columns. - @samukweku Issue #1333
+-   [ENH] `truncate_datetime` now uses a vectorized option. -@samukweku #1337
```
Comment (ericmjl) [inline CHANGELOG.md]: @samukweku is this changelog fully accurate? If not, could you help flesh it out please? To get things started, I'll do a quick run of gitbot to get a summary out for you.

**Comment by ericmjl** [comment]
# GitBot Summary of Changes

This pull request includes several changes across multiple files:

1. In `CHANGELOG.md`, a new enhancement was added to the list, stating that `truncate_datetime` now uses a vectorized option.

2. In `janitor/functions/change_type.py`, the `applymap` function was replaced with `map` for changing the data type of a DataFrame column.

3. In `janitor/functions/pivot.py`, the `is_categorical_dtype` function was replaced with `isinstance(entry, pd.CategoricalIndex)` for checking if an entry is a categorical index.

4. In `janitor/functions/sort_column_value_order.py`, the `future.no_silent_downcasting` option was set to True.

5. In `janitor/functions/truncate_datetime.py`, the function was refactored to use a vectorized option for truncating datetime.

6. In `pyproject.toml`, the configuration for the ruff tool was updated.

7. In `tests/conftest.py`, a new environment variable `PYARROW_IGNORE_TIMEZONE` was set to 1.

8. In `tests/functions/test_complete.py`, the test cases were updated to sort the DataFrame by 'Taxon' and ignore the index.

9. In `tests/functions/test_groupby_agg.py`, the aggregation function was changed from `np.mean` to `\"mean\"`.

10. In `tests/functions/test_groupby_topk.py`, the test cases were updated to sort the DataFrame by columns.

11. In `tests/functions/test_pivot_longer.py`, the `stack` function was updated to use the `future_stack` parameter.

12. In `tests/functions/test_select_columns.py`, the frequency of the date range was changed from 'BM' to 'BME'.

13. In `tests/functions/test_truncate_datetime.py`, the test case was updated to convert the datetime to nanoseconds.

14. In `tests/timeseries/test_fill_missing_timestamps.py`, the frequency of the date range was changed from '1H' to '1h'.

15. In `tests/timeseries/test_flag_jumps.py`, the frequency of the date range was changed from '1H' to '1h'.

cc: @samukweku, please check for correctness!

**Review by ericmjl — APPROVED** [review]

**Comment by samukweku** [comment]
@ericmjl the summary is spot on! Should I update changelog with the summary? or should i trim it?

**Comment by ericmjl** [comment]
> @ericmjl the summary is spot on! Should I update changelog with the summary? or should i trim it?

Hey @samukweku! I think it can be trimmed. What I noticed is that the changelog had some differences compared to the actual changes made, which prompted my comment on the changelog. If you could update just the changelog line entry, then we're good to merge!

**Comment by ericmjl** [comment]
@samukweku, please feel free to do the honors of merging!

**Pull request merged**